src/txmempool.cpp

```diff
@@ -2024,23 +2024,27 @@ bool CCoinsViewMemPool::GetSidechain(const uint256& scId, CSidechain& info) cons
         // Update sidechain info with data from the unconfirmed certificates
         // This is useful for non-ceasing sidechains only as they can have
         // certificates of later epochs.
-        if (info.isNonCeasing() && mempool.mapSidechains.at(scId).mBackwardCertificates.size() > 0)
-        {
-            const uint256& topQualHash = mempool.mapSidechains.at(scId).GetTopQualityCert()->second;
-            const CScCertificate& certTopQual = mempool.mapCertificate[topQualHash].GetCertificate();
-            const auto map_it = mapCumtreeHeight.find(certTopQual.endEpochCumScTxCommTreeRoot.GetLegacyHash());
-            
-            if (map_it == mapCumtreeHeight.end())
-            {
-                LogPrint("mempool", "%s():%d - could not find referenced block for certTopQual %s\n", __func__, __LINE__, certTopQual.GetHash().ToString());
-                info.lastUnconfirmedReferencedHeight = -1; // -1 stands for no valid values in mempool
-            }
-            else
+        if (!mempool.mapSidechains.at(scId).mBackwardCertificates.empty()) {
```

Comment:
Just a few lines above we do 
```
(mempool.mapSidechains.at(scId)
```
Maybe is worth to save a reference and avoid a second lookup on the map

Reply:
Yeah you're right, nice catch! I've reworked this section to remove all the unneeded accesses, thanks (7e39749d9b01b1ee9557e5bc221d95963163f968)